zebra-chain/src/sapling/tests/prop.rs

```diff
@@ -2,33 +2,51 @@ use proptest::prelude::*;
 
 use crate::{
     block,
-    sapling::{self, PerSpendAnchor},
+    sapling::{self, PerSpendAnchor, SharedAnchor},
     serialization::{ZcashDeserializeInto, ZcashSerialize},
     transaction::{LockTime, Transaction},
 };
 
 use futures::future::Either;
 
 proptest! {
-    // TODO: generalise this test for `ShieldedData<SharedAnchor>` (#1829)
+
+    // Serialize and deserialize `PerSpendAnchor` and `SharedAnchor` shielded data
+    //  by including them in V4 and V5 transactions respectivly.
     #[test]
-    fn shielded_data_roundtrip(shielded in any::<sapling::ShieldedData<PerSpendAnchor>>()) {
+    fn shielded_data_roundtrip(
```

Comment:
Please split this test into separate functions for v4 and v5. That way, it's easier to see which version is failing.